src/utils/form/form-control.ts

```diff
@@ -75,17 +81,37 @@ export class FormControl<T> extends AbstractControl {
         this._readonly = isReadonly;
     }
 
+    /**
+     * This specify the ennd of a edition context. (a blur event)
+     */
     public endEdition(): void {
         this._touched = true;
         super.endEdition();
     }
 
-    public reset(): void {
+    /**
+     * Reset the field to it's orginal pristine state.
+     *
+     * @param {T} [initialValue]  a new initial value for the field
+     */
+    public reset(initialValue?: T): void {
```

Comment:
je ne suis pas certain si ce param est nécessaire avec le setter sur initial value ? j'ai l'impression que ca dénature de faire un reset qui change la valeur initiale

Reply:
dans brio on reload nos formulaires systématiquement à chaque ouverture de modale.  On a 2 choix si la valeur change =>

1- On recréé tout le formulaire à chaque fois
2- On rend la valeur initiale "réactive" en passant un callback au lieu d'une valeur hardcodée.

Note à part : Si le portal destroyait correctement son contenu quand on ferme la modale, on aurait à faire ni un ni l'autre :-)

Reply:
avec reset(initialValue) le champs redevient pristine, les validations sont effacés et la valeur intiale est resetté,

C'est différent de formcontrol.initalValue qui lui ne reset pas le champs donc le value du champs reste tant que le formcontrol.reset n'est pas appelé. Donc ce paramêtre optionel répond a votre cas d'utilisation non?

Reply:
le point de @jsroy19 est valide malgré tout.

Reply:
d'accord
mais c'est plus rapide de faire
```
controlxyz.reset('patate');
```
que 
```
controlxyz.initialValue = 'patate'
controlxyz.reset();
```

Reply:
je pense que la rapidité ne devrait pas être un argument au dessus du design. anyway, je ne suis pas non plus offusqué de voir le initialValue dans le reset.

Reply:
Si on fait que le initialValue est un callback dans le setup du formulaire on a aucune assignation à faire et pas de valeur a passer au reset, just sayin ;).  C'est encore plus rapide.

Reply:
Pas mal moins error prone aussi.
Moins possible d'oublier.

Reply:
Ouin, mais est-ce que le champ va devenir pristine? Est-ce que les validations vont être déclenché? Est-ce qu'on veut que les validations soient déclenché?

Il y a aussi le cas où tu attends après le backend pour populer un champs. Au début, la initialValue est vide et après tu veux la changer, mais tu veux aussi la changer avec la valeur.

Reply:
> Si on fait que le initialValue est un callback dans le setup du formulaire on a aucune assignation à faire et pas de valeur a passer au reset, just sayin ;). C'est encore plus rapide.

le probleme que j'ai lorsqu'on fait des unions de type du genre initalValue : T | () => T c'est qu'on souvent obligé de faire des if un peu partout ou tu utilises la valeur T du genre :

`if(T instance of Function) then ... else ...`

Peux-être c'est mon background en Java qui me fait moins aimer ça, mais j'essaie d'éviter ce genre de conditonal le plus possible dans un type system

Reply:
constructor (initialValue: string | function) {
    this.initialValueCallback = typeof initialValue === 'function' : initialValue : () => initialValue
}

Faut juste que tu l'assigne une fois comme du monde et après tu oublie

Reply:
si on est pour ramener la fonction, je préfère quelle soit typée () => T